lib/setExtensions.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * A set of useful extensions to the Set prototype.
+ */
+
+Set.prototype.union = function(other) {
```

Comment:
See my comment below about ensuring dependency ordering. I'm not sure that a Set is what we want anyway, given that order is very important for dependencies. An array with very clear order semantics feels more natural to me.

Although it's true that JS guarantees Set traversal order is insertion order, so maybe this is also fine? I guess that would simplify the pseudocode below to:

```text
allDependencies = Set()
for each element on page:
    for each dependency in element.dependencies:
        allDependencies.add(dependency) // will only add if not already present
```

And the Set insertion-ordering guarantee is want we want.